lib/postprocessing/image.js

```diff
@@ -259,6 +260,10 @@ async function needsImagePostProcess(rendition, source) {
     } else if (!hasJpegQuality(metadata, instructions)) {
         console.log("running post-processing to set jpg quality");
         return true;
+
+    } else if (metadata.format === "SVG") {
+        console.log("running post-processing to convert SVG");
+        return true;
```

Comment:
This isn't necessary as the `isSupportedInputFormat()` already covers it.

Reply:
If I read the logic correctly, checking isSupportedInputFormat() doesn't set post processing to true, checking isSupportedInputFormat() only throws and leaves the conditional block it's in, then proceeds to check if post processing is warranted in a separate conditional.


```
async function needsImagePostProcess(rendition, source) {

.......

    if (!isSupportedInputFormat(metadata.format)) {
        console.log(`Error: failing post-processing because intermediate rendition format '${metadata.format}' is not supported. The worker callback might have generated an incorrect result.`);
         throw new RenditionFormatUnsupportedError(`Format ${instructions.fmt} is not supported for ${source.extension} files`);
    }

    // check if there is an instruction that definitely needs post processing
    // fast check first, which does not requiring reading image
    if (hasExtraInstructions(instructions)) {
        console.log("running post-processing because instructions require it");
        return true;

    } else if ()....

.......

    console.log("skipping post-processing because intermediate rendition meets the target");
    return false;
}
```


```
function isSupportedInputFormat(format) {
    return SUPPORTED_INPUT_FORMAT_IMAGEMAGICK.includes(format);
}
```

Reply:
Yeah, sorry the combination of `isSupportedInputFormat()` and `isTargetFormat()` (line 252) checks should do it.

`isTargetFormat()` checks if the input (source or intermediate) is already in the output (target) format (no post processing), or not (post processing). This should always give post processing if e.g. svg -> png is being handled.